### PR TITLE
crystal: 0.32.1 -> 0.33.0

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
ZHF: #80379
Can't be backported to 20.03 because of feature freeze (0.32.1 build for 20.03 handled in #80389)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/80390)

### CODE REVIEW

**Comment by filalex77** [comment]
@GrahamcOfBorg build crystal

**Comment by filalex77** [comment]
@GrahamcOfBorg build crystal

**Comment by filalex77** [comment]
Sorry, accidentally committed a (now revoked) secret.

@GrahamcOfBorg build crystal

**Comment by filalex77** [comment]
@GrahamcOfBorg build crystal

File: pkgs/development/compilers/crystal/default.nix [hunk 1]
```diff
@@ -102,6 +98,8 @@ let
 
     makeFlags = [
       "CRYSTAL_CONFIG_VERSION=${version}"
+      # FIXME: not sure if this is the best idea
```
Comment (yegortimoshenko) [inline pkgs/development/compilers/crystal/default.nix]: Why not? LGTM. The alternative is (I guess) to set writable `HOME`.

File: pkgs/development/compilers/crystal/default.nix [hunk 2]
```diff
@@ -271,7 +278,13 @@ in rec {
     binary = binaryCrystal_0_31;
   };
 
-  crystal = crystal_0_32;
+  crystal_0_33 = generic {
+    version = "0.33.0";
+    sha256  = "1zg0qixcws81s083wrh54hp83ng2pa8iyyafaha55mzrh8293jbi";
```
Comment (yegortimoshenko) [inline pkgs/development/compilers/crystal/default.nix]:
```suggestion
    sha256 = "1zg0qixcws81s083wrh54hp83ng2pa8iyyafaha55mzrh8293jbi";
```

**Review by lukateras — APPROVED** [review]
Builds, compiler is working.

**Comment by yegortimoshenko** [comment]
Will merge on Feb 25 12 pm UTC unless anyone minds otherwise.

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build crystal_0_27 crystal_0_29 crystal_0_30 crystal_0_31 crystal_0_32 crystal_0_33

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build crystal_0_32

**Comment by yegortimoshenko** [comment]
Unfortunately, looks like changes to `generic` break `crystal_0_32_0` build :(

**Comment by peterhoeg** [comment]
Superseded by #85432

**Pull request closed without merging**